airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_40_12_001__AddWebhookOperationColumns.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.instance.configs.migrations;
+
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.jooq.DSLContext;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO: update migration description in the class name
+public class V0_40_12_001__AddWebhookOperationColumns extends BaseJavaMigration {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(V0_40_12_001__AddWebhookOperationColumns.class);
+
+  @Override
+  public void migrate(final Context context) throws Exception {
+    LOGGER.info("Running migration: {}", this.getClass().getSimpleName());
+
+    // Warning: please do not use any jOOQ generated code to write a migration.
+    // As database schema changes, the generated jOOQ code can be deprecated. So
+    // old migration may not compile if there is any generated code.
+    final DSLContext ctx = DSL.using(context.getConnection());
+    addWebhookOperationConfigColumn(ctx);
+    addWebhookOperationType(ctx);
+    addWebhookConfigColumnsToWorkspaceTable(ctx);
+  }
+
+  private void addWebhookConfigColumnsToWorkspaceTable(final DSLContext ctx) {
+    ctx.alterTable("workspace")
+        .addColumnIfNotExists(DSL.field(
+            "webhook_operation_configs",
+            SQLDataType.JSONB.nullable(true)))
+        .execute();
+  }
+
+  private void addWebhookOperationType(final DSLContext ctx) {
+    ctx.alterType("operator_type").addValue("webhook").execute();
+  }
+
+  private void addWebhookOperationConfigColumn(final DSLContext ctx) {
+    ctx.alterTable("operation").addColumnIfNotExists(DSL.field("operator_webhook",
```

Comment:
Basically: different connections might want different configs (and that happens by creating different entries in the `operation` table).

There are a few reasons:
- For one, we need to say "use the webhook for _this_ connection (and not _that_ connection)". So we need to create an operation, and then the connection has a list of operations to execute, so we can indicate that we want to use the operation with a particular connection.
- We might also want to e.g., change the parameters between connections.

Reply:
That sounds like something better suited for the `connection_operation` table. Maybe worth a quick sync tmrw?

Reply:
The `connection_operation` table today just holds:
```
- id
- connection_id
- operation_id
- created_at/updated_at
```
essentially just hooking up connections and operations. (I'm not 100% sure why the `connection_operation` table exists, instead of just having an `operation_ids` column on the `connection` table, but I assume there's a good reason).

Whereas the `operation` table is where e.g., the existing `dbt` transformation config lives (which is mostly how I decided to put it there).

Regardless, +1 to syncing tomorrow - maybe good to run through the whole thing e2e for some early feedback as well.

Reply:
spoke offline: this is column is needed so webhook configs are stored on the operation table.